docs/guide/dynamic_link.md

```diff
@@ -0,0 +1,70 @@
+# Dynamic linking
+
+If you want to support dynamic linkages between python projects or system
+libraries, you will likely encounter some issues in making sure the compiled
+libraries/python bindings work after the wheel is created and the python project
+is installed on the system. The most common issue are the missing hints pointing
+to where the runtime libraries are located, specifically `RPATH` on Linux and
+MacOS systems, and `PATH`/`os.add_dll_directory` on Windows systems. Here are
+some recommendations on how to address them.
+
+## Link to the static libraries
+
+The easiest solution is to make sure you link to the static libraries
+counterparts during the CMake build. How to achieve this depends on the specific
+dependency, how it is imported in the CMake project and how the dependency is
+packaged in each ecosystem.
+
+For example for [Boost][FindBoost] this is controlled via the variable
+`Boost_USE_STATIC_LIBS`.
+
+[FindBoost]: inv:cmake:cmake:module#module:FindBoost
+
+## Wheel repair tools
+
+There are wheel repair tools for each operating system that bundle any dynamic
+libraries used and patch the libraries/python bindings to point to prioritize
+those libraries. The most common tools for these are [auditwheel] for Linux,
+[delocate] for MacOS and [delvewheel] for Windows. [cibuildwheel] incorporates
+these tools in its [repair wheel] feature.
+
+One downside of these tools is that all dependencies are duplicated and bundled
+in each project.
```

Comment:
These tools also do name mangling, which ensures that each wheel is independent, as required to be a valid manylinux wheel. Wheels are not allowed to have binary dependencies on each other. This irritates some packages like `scipy` and `numpy`, which would love to be able to share thread pools, for example, but that isn't allowed. It is allowed with Conda.

Reply:
Hmm, any reading that I can look at? The only reference I found is about avoiding global libraries and clashes, but not on binary dependence. I've only looked at PEP600 and PEP513.

Yeah WheelNext would be great. Any indication if this would be available for build-backends to experiments first or will it need pip install frontends interface first?

Yeah more caveats would be in order for this section. I am thinking that this would be useful when the dependent package has a good SOVERSION matching with the pyproject file and the consumer has a good version pinning. Yeah if you do a `pip upgrade` of the dependency it would potentially break things, but that's true for pure python as well.